### PR TITLE
WIP - add more styles

### PR DESCRIPTION
**Pull request by bwinton**
This gets it closer to Shorlander's mockup.  Any suggestions on how to improve it are welcome!

for reference https://people-mozilla.org/~shorlander/projects/photon/Mockups/windows-10.html

### CODE REVIEW

**Review by victorporof — COMMENTED** [review]
Couple of notes:

With these changes, it's impossible to move the window around after filling the tabbar with tabs.

Filling the tabbar with tabs overflows the window minimize button on Windows and Linux.

On macOS, window controls aren't vertically centered in the tabbar anymore (since it's shorter). Should either make the tabbar taller, or nudge the native controls a little more by using a "onlyTitleBarHidden" window instead of "onlyTitleBarHiddenAndWindowControlsInset".

The window border doesn't respect the fact that windows are curved on macOS, so corners have a few pixels cut.

**Review by victorporof — CHANGES_REQUESTED** [review]
This is great overall, but there's a few small oddities you may want to address.

**Review by ericawright — COMMENTED** [review]
Some nits
- could you add the top border to the active tab after the background animation is finished, or slide it up with the animation?
- on Shorlander's mock there is a version with extra drag space.

missing from this:
- More padding/margin on the right side of the close button.
- grey top border with growing animation when hovering over inactive tab
- hover effect on close button

**Comment by bwinton** [comment]
I asked Stephen about this, and he said that I should use the version without any of the options checked, so no extra drag space in this version.  (Perhaps I'll add an about:visuals options page later to let us toggle stuff.  😉 )

**Comment by victorporof** [comment]
@bwinton That would be a great exercise for us to 1. Build some about pages and 2. Have some kind of prefs story.

**Comment by victorporof** [comment]
Is this ready for re-review?

**Comment by ericawright** [comment]
Almost. Adding a few more changes first. I'll comment here once it's ready.

**Comment by ericawright** [comment]
@bwinton please test the changes on windows. @victorporof ready for re-review :)

**Comment by bwinton** [comment]
(I made some minor tweaks on Windows, but other than those, it looks good.)

File: src/browser-frontend/views/browser/chrome/tabbar.css [hunk 1]
```diff
@@ -15,16 +15,17 @@ specific language governing permissions and limitations under the License.
   height: var(--theme-tabbar-height);
   box-sizing: border-box;
   flex: 1;
-  padding: 0 10px;
+  padding: 0 10px 0 40px;
```
Comment (victorporof) [inline src/browser-frontend/views/browser/chrome/tabbar.css]: Any reason why we need both margins and paddings (here and below) to make the tabbar not overflow over window controls? Why not just use one?

**Review by victorporof — APPROVED** [review]
Note that when the blue flash tab loading animations are displayed, the selected tab's top blue bar disappears. Your choice if you want to fix that here or in a later PR.

Window controls are still not vertically aligned on macOS, as per https://github.com/victorporof/tofino/pull/96#pullrequestreview-36483871 Again, your choice if you want to address this in this PR or not, but have you tried looking into it? 

Some more style nits below.

File: src/browser-frontend/views/browser/chrome/tabbar/new-tab-button.jsx [hunk 2]
```diff
@@ -43,9 +43,6 @@ export default class NewTabButton extends PureComponent {
           height="12px"
           styleName="icon"
         />
-        <div>
-          New Tab
-        </div>
```
Comment (victorporof) [inline src/browser-frontend/views/browser/chrome/tabbar/new-tab-button.jsx:48]: With this change, the new tab button contents aren't properly horizontally aligned in the button.
screen shot 2017-05-12 at 2 28 53 pm

Comment (bwinton) [reply src/browser-frontend/views/browser/chrome/tabbar/new-tab-button.jsx:48]: Fixed.

File: src/browser-frontend/views/browser/chrome/tabbar.css (hunk 1 above)
Comment (bwinton) [reply src/browser-frontend/views/browser/chrome/tabbar.css]: Fixed.

**Comment by bwinton** [comment]
> Note that when the blue flash tab loading animations are displayed, the selected tab's top blue bar disappears. Your choice if you want to fix that here or in a later PR.

New PR.  (It looks like we'll need a new gif from Eric…)

> Window controls are still not vertically aligned on macOS, as per #96 (review) Again, your choice if you want to address this in this PR or not, but have you tried looking into it?

We did, and this is hella-hard to fix.  Possibly impossible, unless we want to draw the macOS windows controls ourselves…

**Pull request merged**